### PR TITLE
Newseed

### PR DESCRIPTION
**Pull request by Peartes**
[x] Stage3 UI
[x] Stage3 Validation

### CODE REVIEW

File: src/newSeed/baseStage.scss [hunk 1]
```diff
@@ -1,66 +1,100 @@
 @import "styles/colors";
 // This styles are common to all the stages components
 // Theme for the stages
-@mixin stageTheme {
-  * {
-    font-size: 1rem;
+.infoContainer {
+  font-size: 1rem;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:5]: Is 1rem not already the default font size?

File: src/newSeed/baseStage.scss [hunk 2]
```diff
@@ -1,66 +1,100 @@
...
+  label {
+    display: block;
+    font-size: 1.1em;
+    line-height: 2em;
+    color: $color-white;
+  }
+  input {
+    width: 26em;
+    height: 2.7em;
+    background-color: #1b1722;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:52]: this is already the default

File: src/newSeed/baseStage.scss [hunk 3]
```diff
@@ -1,66 +1,100 @@
...
+    border: 1px solid $color-darkpink-2;
+    border-radius: 0.6em;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:54]: should be 10px according to what i see in the mock.  I think we should use 10px.

File: src/newSeed/baseStage.scss [hunk 4]
```diff
@@ -1,66 +1,100 @@
...
+    font-size: 1.1em;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:55]: Should be 18px

File: src/newSeed/baseStage.scss [hunk 5]
```diff
@@ -1,66 +1,100 @@
...
+    color: #989798;
+    font-weight: 200;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:57]: should be "normal"

File: src/newSeed/baseStage.scss [hunk 6]
```diff
@@ -1,66 +1,100 @@
...
+    .cancel {
+      cursor: pointer;
+      color: $color-orange;
+      font-family: Inter;
     }
-    a {
-      color: #989798;
-      text-decoration: underline;
+  }
+  .stageButton {
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:39]: shouldn't this be inside of .buttonContainer or else be removed?  Doesn't make sense to me.

File: src/newSeed/baseStage.scss [hunk 7]
```diff
@@ -1,66 +1,100 @@
...
+    margin-bottom: 5em;
+  }
+  .infoInput {
+    margin-bottom: 2em;
+  }
+  .buttonContainer {
+    display: flex;
+    flex-flow: row wrap;
+    column-gap: 3em;
+    font-weight: bold;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:31]: remove this.  the button styles are already set globally.  if it is for ".cancel" then put it there.

File: src/newSeed/baseStage.scss [hunk 8]
```diff
@@ -1,66 +1,100 @@
...
+  input {
+    width: 26em;
+    height: 2.7em;
+    background-color: #1b1722;
+    border: 1px solid $color-darkpink-2;
+    border-radius: 0.6em;
+    font-size: 1.1em;
+    color: #989798;
+    font-weight: 200;
+    font-family: Inter;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:58]: already set globally

File: src/newSeed/baseStage.scss [hunk 9]
```diff
@@ -1,66 +1,100 @@
...
+  }
+  /* Chrome, Safari, Edge, Opera */
+  input::-webkit-outer-spin-button,
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:61]: what are these for?

File: src/newSeed/baseStage.scss [hunk 10]
```diff
@@ -1,66 +1,100 @@
...
+  input::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+    margin: 0;
+  }
+  /* Firefox */
+  input[type="number"] {
+    -moz-appearance: textfield;
+  }
+  a {
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:70]: where is this used?

File: src/newSeed/baseStage.scss [hunk 11]
```diff
@@ -1,66 +1,100 @@
...
+    width: 25em;
+    height: 4.5em;
+    border-radius: 0.625em;
+    border: 1px solid $color-darkpink-2;
+    background-color: #1b1722;
+    font-weight: 200;
+    color: #989798;
+  }
 }
 .page.stageContainer {
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:95]: I think stageContainer needs to be removed in newseed.html

File: src/newSeed/baseStage.scss [hunk 12]
```diff
@@ -87,6 +121,9 @@
           .deleteLink {
             cursor: pointer;
             justify-self: flex-end;
+            i {
+              font-family: "Font Awesome\ 5 Free";
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:125]: setting this font-family should not be necessary.  what does it accomplish here?

File: src/newSeed/seedConfig.ts [hunk 13]
```diff
@@ -14,9 +15,38 @@ export interface IProjectDetails {
   category: string,
 }
 
+export interface ITokenDetails {
+  fundingAddress: string,
+  fundingTicker: string,
```
Comment (dkent600) [inline src/newSeed/seedConfig.ts:20]: change to Symbol

File: src/newSeed/seedConfig.ts [hunk 14]
```diff
@@ -14,9 +15,38 @@ export interface IProjectDetails {
...
+  fundingIcon: string,
+  seedAddress: string,
+  seedTicker: string,
```
Comment (dkent600) [inline src/newSeed/seedConfig.ts:23]: change to Symbol

File: src/newSeed/stage2.html [hunk 15]
```diff
@@ -40,13 +43,16 @@
...
-            maxlength="150">
+            maxlength="150"
+          >
           </textarea>
         </div>
       </div>
       <div class="buttonContainer">
         <div class="stageButton cancel" click.delegate="back()">PREVIOUS</div>
-        <div class="stageButton proceed button1" click.delegate="proceed()">NEXT</div>
+        <div class="stageButton proceed button1" click.delegate="proceed()">
```
Comment (dkent600) [inline src/newSeed/stage2.html:53]: restore original formatting

File: src/newSeed/stage3.html [hunk 16]
```diff
@@ -1,5 +1,95 @@
...
+              </div>
+              <div  class="infoInput supply" value.bind="distrib.amount">
+                <numeric-input type="number" value.bind="distrib.amount" placeholder="0" class="numberInput"></numeric-input>
+              </div>
+              <div  class="infoInput supply numberInput" value.bind="distrib.lockup">
+                <input type="number" value.bind="distrib.lockup" placeholder="0" class="numberInput">
+              </div>
+            </div>
+            </template>
+            <div
```
Comment (dkent600) [inline src/newSeed/stage3.html:81]: put this on one line

File: src/newSeed/stage3.ts [hunk 17]
```diff
@@ -1,5 +1,87 @@
...
 export class Stage3 extends BaseStage {
+  constructor(
+    eventAggregator: EventAggregator,
+    private tokenService: TokenService,
+    router: Router,
+  ) {
+    super(router, eventAggregator);
+  }
+  addTokenDistribution(index:number): void {
+    if (index === -1) {
```
Comment (dkent600) [inline src/newSeed/stage3.ts:18]: this looks like a no-op

File: src/newSeed/stage3.ts [hunk 18]
```diff
@@ -1,5 +1,87 @@
...
+    }
+    else if (!this.seedConfig.tokenDetails.initSupply || this.seedConfig.tokenDetails.initSupply.lte(0)) {
+      message = "Please enter a non-zero number for Initial Supply";
+    }
+    // Check the token distribution
+    this.seedConfig.tokenDetails.tokenDistrib.forEach((tokenDistrb: {category: string, amount: BigNumber, lockup: number}) => {
+      if (!tokenDistrb.category) {
+        message = "Please enter a value for Category";
+      } else if (!tokenDistrb.amount || tokenDistrb.amount.lte(0)) {
+        message = `Please enter a non-zero number for category ${tokenDistrb.category} Amount`;
```
Comment (dkent600) [inline src/newSeed/stage3.ts:45]: Category with a capital 'C'?

File: src/newSeed/stage3.ts [hunk 19]
```diff
@@ -1,5 +1,87 @@
...
+    } else {
+      this.stageState[this.stageNumber].verified = true;
+      this.next();
+    }
+  }
+  // TODO: Add a loading comp to the view while fetching
+  getTokenInfo(type: string): void {
+    if (type === "fund" && this.seedConfig.tokenDetails.fundingAddress) {
+      this.tokenService.getTokenInfoFromAddress(this.seedConfig.tokenDetails.fundingAddress).then((tokeInfo: ITokenInfo) => {
+        if (tokeInfo.symbol !== "N/A") {
```
Comment (dkent600) [inline src/newSeed/stage3.ts:62]: Would be better to use terniary logic:

this.seedConfig.tokenDetails.fundingTicker = (tokeInfo.symbol !== "N/A") ? tokeInfo.symbol  : undefined;

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/newSeed/baseStage.scss (hunk 7 above)
Comment (Peartes) [reply src/newSeed/baseStage.scss:31]: This is for the container holding the buttons and not for the buttons themselves. It is needed.

File: src/newSeed/baseStage.scss (hunk 8 above)
Comment (Peartes) [reply src/newSeed/baseStage.scss:58]: The text color for these inputs are not white as set in the global style. I'll keep there here

File: src/newSeed/baseStage.scss (hunk 9 above)
Comment (Peartes) [reply src/newSeed/baseStage.scss:61]: Remove the number input spinner that displays next to a webkit based browser see the attachment
Screenshot 2021-05-02 at 23 34 15

File: .vscode/settings.json [hunk 20]
```diff
@@ -6,5 +6,6 @@
   "[typescript]": {
     "editor.formatOnSave": false
   },
-  "files.eol": "\n"
+  "files.eol": "\n",
+  "editor.wordWrapColumn": 100
```
Comment (dkent600) [inline .vscode/settings.json:10]: Can you please remove this line (wordWrapColumn)

Comment (dkent600) [inline .vscode/settings.json:10]: If this is a problem, then we should consider removing settings.json from the repo

File: src/newSeed/baseStage.scss (hunk 7 above)
Comment (dkent600) [reply src/newSeed/baseStage.scss:30]: what is in the container except buttons?

File: src/newSeed/baseStage.scss (hunk 10 above)
Comment (dkent600) [reply src/newSeed/baseStage.scss:67]: ?

File: src/newSeed/stage2.html (hunk 15 above)
Comment (dkent600) [reply src/newSeed/stage2.html:53]: ?

File: src/newSeed/stage3.html (hunk 16 above)
Comment (dkent600) [reply src/newSeed/stage3.html:81]: ?

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 20 above)
Comment (dkent600) [reply .vscode/settings.json:10]: If this is a problem, then we should consider removing settings.json from the repo

File: src/newSeed/baseStage.scss (hunk 7 above)
Comment (Peartes) [reply src/newSeed/baseStage.scss:30]: Nothing but buttons but there are 2 buttons and the container is for me to style how they are positioned in the container

Comment (dkent600) [reply src/newSeed/baseStage.scss:30]: position is fine, but this comment is on line 30 which is font-weight, which you don't need here

File: src/newSeed/baseStage.scss (hunk 10 above)
Comment (Peartes) [reply src/newSeed/baseStage.scss:67]: Oh yes, this is for the 3rd stage where I could not use the numeric-input. Setting the type of the input field to number adds a spinner to the side of the input field. This style is to remove that spinner for webkit browsers

File: .vscode/settings.json (hunk 20 above)
Comment (Peartes) [reply .vscode/settings.json:10]: I don't thin it would be

File: src/newSeed/baseStage.scss (hunk 7 above)
Comment (Peartes) [reply src/newSeed/baseStage.scss:30]: Aha, didn't catch that

File: .vscode/settings.json [hunk 21]
```diff
@@ -6,5 +6,7 @@
   "[typescript]": {
     "editor.formatOnSave": false
   },
-  "files.eol": "\n"
+  "files.eol": "\n",
+  "editor.wordWrapColumn": 150,
+  "html.format.wrapLineLength": 0
```
Comment (dkent600) [inline .vscode/settings.json:11]: Please undo this change.  If that is a problem, then we should remove settings.json from the repo.

File: src/newSeed/baseStage.scss [hunk 22]
```diff
@@ -1,66 +1,96 @@
...
+    color: $color-white;
+  }
+  input {
+    width: 25em;
+    height: 2.7em;
+    border: 1px solid $color-darkpink-2;
+    border-radius: 16px;
+    font-size: 18px;
+    color: #989798;
+    font-weight: normal;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:54]: this is not needed

File: src/newSeed/baseStage.scss [hunk 23]
```diff
@@ -1,66 +1,96 @@
...
+    line-height: 2em;
+    color: $color-white;
+  }
+  input {
+    width: 25em;
+    height: 2.7em;
+    border: 1px solid $color-darkpink-2;
+    border-radius: 16px;
+    font-size: 18px;
+    color: #989798;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:53]: should be #1b1722 ($BG01) (see latest _colors.scss)

File: src/newSeed/baseStage.scss [hunk 24]
```diff
@@ -1,66 +1,96 @@
...
+  label {
+    display: block;
+    font-size: 1.1em;
+    line-height: 2em;
+    color: $color-white;
+  }
+  input {
+    width: 25em;
+    height: 2.7em;
+    border: 1px solid $color-darkpink-2;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:50]: Should be #403453 ($Border01) (see latest _colors.scss)

File: src/newSeed/baseStage.scss [hunk 25]
```diff
@@ -1,66 +1,96 @@
...
+    border-radius: 16px;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:51]: should be 10px

File: src/newSeed/baseStage.scss [hunk 26]
```diff
@@ -1,66 +1,96 @@
...
     }
   }
+  textarea {
+    resize: none;
+    width: 25em;
+    height: 4.5em;
+    border-radius: 0.625em;
+    border: 1px solid $color-darkpink-2;
+    background-color: #1b1722;
+    font-weight: normal;
```
Comment (dkent600) [inline src/newSeed/baseStage.scss:87]: not needed

File: src/newSeed/stage1.html [hunk 27]
```diff
@@ -33,41 +33,48 @@
...
-          click.delegate="addCustomLinks(-1)">
+          click.delegate="addCustomLinks(-1)"
+        >
           <a>+ Add Social Media link</a>
         </div>
       </div>
       <div class="buttonContainer">
         <div class="stageButton cancel" click.delegate="cancel()">CANCEL</div>
-        <div class="stageButton proceed button1" click.delegate="proceed()">NEXT</div>
+        <div class="stageButton proceed button1" click.delegate="proceed()">
```
Comment (dkent600) [inline src/newSeed/stage1.html:75]: Please revert to the original single line

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 28]
```diff
@@ -6,5 +6,7 @@
   "[typescript]": {
     "editor.formatOnSave": false
   },
-  "files.eol": "\n"
+  "files.eol": "\n",
+  "editor.wordWrapColumn": 150,
```
Comment (dkent600) [inline .vscode/settings.json:10]: I am confused.  I asked you to remove line 10, but you added line 11, which just makes it worse.

File: src/newSeed/baseStage.scss (hunk 1 above)
Comment (dkent600) [reply src/newSeed/baseStage.scss]: This is the global default font size.  this line should be removed

File: src/newSeed/baseStage.scss (hunk 26 above)
Comment (dkent600) [reply src/newSeed/baseStage.scss:84]: still there, should be deleted.

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 28 above)
Comment (Peartes) [reply .vscode/settings.json:10]: Hmm the request said undo change to this line, that confused me

**Review by dkent600 — APPROVED** [review]
